feature/nesworld-scraper

cwallace3421 opened the pull request:
## nesworld scraper built in Node JS

Includes:
- The nesworld scraper.
- The entries produced by the new scraper.

**Result is 91 new entries.**
If I should make this two pull requests, one for the scraper and one for the entries then let me know. 

I think there are some duplicate entries with different names, but not sure the best way to sort through them rather than doing it manually.

I updated the scraper to use the new format/scheme you started using recently. With the files array and the hashes for each file. Let me know if it looks correct!

The README for the scraper includes the results of the scrape and what entries don't have screenshots etc.

avivace commented:
Brilliant work! This looks good. You've also adopted the new schema, making things easy.
I've merged the scraper since it looks good, works and it's documented.

I'd wait to merge the entries.

We need to:

- Remove lsdj
- Remove duplicate entries but with different versions (maybe merge them or retain only the last one?) like `jetpak-dx-beta-version-001","jetpak-dx-beta-version-002","jetpak-dx-beta-version-003","jetpak-dx-beta-version-004","jetpak-dx-beta-version-005","jetpak-dx-beta-version-006","jetpak-dx-beta-version-007","jetpak-dx-beta-version-008`.
- If the screenshot is not present, `placeholder.png` shouldn't be committed and the `screenshot` array should simply be absent or empty. It's up to the frontend or the API consumer to use the placeholder when there are no screenshot available
- Everything is an homebrew, but the `homebrew` tag is actually used to refer to "tools" and generally not games or demos. This was my bad as it's not mentioned in the schema definition (look at https://gbhh.avivace.com/, we have an example of what is an "homebrew"). We may change this, however, every entry from nesdev is a game.

cwallace3421 commented:
Made the changes you asked:
- There was a 1.0 version of `jetpak` so kept that, and removed the rest.
- There were multiple versions of `mgb` as well so I did the same, just kept the latest version.
- Updated the gameList.json.
- Removed the `placeholder.png` from the screenshot array if it was there (just leaving an empty array), and removing the `placeholder.png` from the filesystem in each entry that had it.
- Set all the nesworld entries to typetag game.

cwallace3421 commented:
Also will need to update the nesworld scraper at some point so it doesn't use the placeholder image etc
